### ISSUE TITLE
After enabling noUncheckedIndexedAccess, using ++ on a default undefined does not trigger an error message.

### ISSUE THREAD

**Issue by LiST-GIT** [microsoft/TypeScript#58235]
### 🔎 Search Terms

noUncheckedIndexedAccess
undefined

### 🕗 Version & Regression Information

5.4.5

### ⏯ Playground Link

https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true#code/G4QwTgBAFjBcEG8IG0DWBTAnvAzgFzAEsA7AcwF15iBXAWwCN1IBfCAXkWYG4AoUSEAEZ20GMgDkIHABNx5XgHoFEAHoB+PuAggATCJhQJU2eQDUpxcvU8bmyPUGxqxaegBmJdNIA+NBk14Hcy4gA

### 💻 Code

I have enabled noUncheckedIndexedAccess.

```ts
var hhh: { [key: string]: number } = {};
var a1 = hhh['asd'];
// ^? var a1 :number | undefined
var a2 = hhh['asd']++; // !!!!! I was expecting an error here, but there wasn't one.
// ^? var a2: number



var b1:undefined|number;
b1++; // 'b1' is possibly 'undefined'.

```

### 🙁 Actual behavior

There was no error when executing `var a2 = hhh['asd']++;`.

### 🙂 Expected behavior

I need an error of `hhh['asd'] is possibly 'undefined'.`

**Comment by MartinJohns** [comment]
Sounds like a duplicate of #53303.

**Comment by snarbies** [comment]
This problem also affects this form: `hhh["asd"] = hhh["asd"]++;` 

I'm confused as to why the linked issue is closed as "Working as intended". The expression `hhh["asd"]++` decomposes to:

```typescript
let temp = hhh["asd"];
temp++;
hhh["asd"] = temp;
```

Written out that way, TS correctly identifies that we are reading a possibly-undefined value and then applying the `++` operator to it, which is an error. How is it correct that the same check should be elided with compound assignment when it represents identical logic?

This is the explanation offered:

>Since it's both a read and a write, why is one of the answers correct and the other not? They have equal standing.

What has equal standing? The order of operations is read, increment, write. It's not one or the other, it's those three, in that order.

**Comment by RyanCavanaugh** [comment]
The linked issue is about what the language service (i.e. tooltips) reports about the left operand's type, not whether or not there's an error when you use `++` on that. It's not a duplicate.
